### PR TITLE
separate get_open_episode_case_from_person into two functions

### PR DESCRIPTION
**Pull request by nickpell**
@proteusvacuum

### CODE REVIEW

File: custom/enikshay/case_utils.py [hunk 1]
```diff
@@ -71,7 +71,19 @@ def get_open_episode_case_from_person(domain, person_case_id):
         raise ENikshayCaseNotFound(
             "Person with id: {} exists but has no open occurrence cases".format(person_case_id)
         )
-    occurrence_case = open_occurrence_cases[0]
+    return open_occurrence_cases[0]
+
+
+def get_open_episode_case_from_person(domain, person_case_id):
```
Comment (nickpell) [inline custom/enikshay/case_utils.py]: oh i see

Comment (nickpell) [inline custom/enikshay/case_utils.py]: updated

File: custom/enikshay/case_utils.py [hunk 2]
```diff
@@ -74,28 +74,40 @@ def get_open_occurrence_case_from_person(domain, person_case_id):
...
     else:
         raise ENikshayCaseNotFound(
-            "Person with id: {} exists but has no open episode cases".format(person_case_id)
+            "Occurrence with id: {} exists but has no open episode cases".format(occurrence_case_id)
         )
 
 
+def get_open_episode_case_from_person(domain, person_case_id):
+    """
+    Gets the first open 'occurrence' case for the person
```
Comment (proteusvacuum) [inline custom/enikshay/case_utils.py]: Gets the first open `episode` case

File: custom/enikshay/case_utils.py [hunk 3]
```diff
@@ -74,28 +74,40 @@ def get_open_occurrence_case_from_person(domain, person_case_id):
     return open_occurrence_cases[0]
 
 
-def get_open_episode_case_from_person(domain, person_case_id):
+def get_open_episode_case_from_occurrence(domain, occurrence_case_id):
     """
     Gets the first open 'episode' case for the person
```
Comment (proteusvacuum) [inline custom/enikshay/case_utils.py]: "for the occurrence"

File: custom/enikshay/case_utils.py [hunk 4]
```diff
@@ -71,19 +71,43 @@ def get_open_episode_case_from_person(domain, person_case_id):
...
+def get_open_episode_case_from_person(domain, person_case_id):
+    """
+    Gets the first open 'occurrence' case for the person
+
+    Assumes the following case structure:
+    Person <--ext-- Occurrence <--ext-- Episode
+
+    """
+    return get_open_episode_case_from_occurrence(
+        get_open_occurrence_case_from_person(person_case_id).case_id
```
Comment (proteusvacuum) [inline custom/enikshay/case_utils.py]: this needs a `domain`.

There's a test [here](https://github.com/dimagi/commcare-hq/blob/f1ee1f44189b9371390becc98312cbd14959da39/custom/enikshay/tests/test_case_utils.py#L73-L74) you can run to validate this is working.

**Review by proteusvacuum — CHANGES_REQUESTED** [review]

File: custom/enikshay/case_utils.py (hunk 2 above)
Comment (nickpell) [inline custom/enikshay/case_utils.py]: updated

File: custom/enikshay/case_utils.py (hunk 4 above)
Comment (nickpell) [inline custom/enikshay/case_utils.py]: oops. fixed, the test passed locally

Comment (nickpell) [inline custom/enikshay/case_utils.py]: tests added

**Review by proteusvacuum — APPROVED** [review]

**Pull request merged**